### PR TITLE
[SET-67] Returning proper inquiries

### PR DESCRIPTION
**Pull request by dallascrichmond**
## 🎯 Summary
- Adds user's guid to inquiries that they have worked on (The ones they create for now)
- Returns only the inquiries that a user has worked on (unless admin)

Use test accounts to see that admins have access to all inquiries while, hm, shr and adm only have access to the ones they work on/ create. The next step is some conditional rendering to only allow for HMs to create the inquiries, but for now you can create inquiries as any role (will be changed in future work)

[SET-67](https://citz-imb.atlassian.net/jira/software/c/projects/SET/boards/51?selectedIssue=SET-67)

## 🔰 Checklist

- [x] I have read and agree with the following checklist and am following the guidelines in our [Code of Conduct](/CODE_OF_CONDUCT.md) document.

> - I have performed a self-review of my code.
> - I have commented my code, particularly in hard-to-understand areas.
> - I have made corresponding changes to the documentation where required.
> - I have tested my changes to the best of my ability.
> - I have consulted with the team if introducing a new dependency.
> - My changes generate no new warnings.

### CODE REVIEW

File: api/express.ts [hunk 1]
```diff
@@ -51,10 +51,10 @@ app.use(
 );
 
 app.use('/', routers.healthRouter);
+app.use('/', protectedRoute(['admin', 'hm', 'shr', 'adm'], { requireAllRoles: false }), routers.userRouter);
```
Comment (BradyMitch) [inline api/express.ts:54]: This won't work. Take a look at https://github.com/bcgov/citz-imb-kc-express/issues/14 and then read the README

File: frontend/src/pages/Home/Home.tsx [hunk 2]
```diff
@@ -14,9 +15,13 @@ const a11yProps = (index: number) => {
 const Home = () => {
   const [value, setValue] = useState(0);
   const { isAuthenticated, hasRole } = useAuthentication();
-  const { InquiryTable } = useInquiry();
   const { UserTable } = useUser();
   const { SalaryDataTable } = useSalaryData();
+  const { state: authState } = useKeycloak();
+  let InquiryParams;
+  const user = authState.userInfo;
```
Comment (BradyMitch) [inline frontend/src/pages/Home/Home.tsx:22]: Same here.

File: api/express.ts (hunk 1 above)
Comment (dallascrichmond) [reply api/express.ts:54]: @BradyMitch Yeah I took a look at that when I ran into this issue. I ended up giving each role access to every endpoint, as it turns out most roles need access anyways. I know the solution was to add "route1", "route2", but I wasn't sure how that would be effected by the proxy considering we had to set it up with the '/'s in the first place. Let me know your thoughts. As it is now, it does seem to work :)

File: frontend/src/hooks/useInquiry/useInquiry.ts [hunk 3]
```diff
@@ -2,15 +2,32 @@
...
 
-export const useInquiry = () => {
-  const endPoint = 'inquiry';
+export const useInquiry = (guid?: string) => {
+  let endPoint = 'inquiry';
+  if (guid) endPoint = `${endPoint}/guid?guid=${guid}`;
   const title = 'Inquiry';
+
+  const { state: authState, hasRole } = useKeycloak();
+  const user = authState.userInfo;
```
[the remark this answers is not in the document]
Comment (dallascrichmond) [reply frontend/src/hooks/useInquiry/useInquiry.ts:18]: @BradyMitch Sweet! I will change that! :)

**Review by BradyMitch — APPROVED** [review]

**Pull request merged**